tests/integration/integration_python_test.go

```diff
@@ -1422,3 +1422,36 @@ func TestParameterizedPython(t *testing.T) {
 		},
 	})
 }
+
+//nolint:paralleltest // ProgramTest calls t.Parallel()
+func TestConfigGetterOverloads(t *testing.T) {
+	t.Parallel()
+	e := ptesting.NewEnvironment(t)
+	defer e.DeleteIfNotFailed()
+	e.ImportDirectory("python/config-getter-types")
+
+	stackName := ptesting.RandomStackName()
+	e.RunCommand("pulumi", "install")
+	e.RunCommand("pulumi", "login", "--cloud-url", e.LocalURL())
+	e.RunCommand("pulumi", "stack", "init", stackName)
+	defer e.RunCommand("pulumi", "stack", "rm", "--yes", "--stack", stackName)
+
+	// ProgramTest installs extra dependencies as editable packages using the `-e` flag, but typecheckers do not
+	// handle editable packages well. We have to manually install the SDK without `-e` flag instead.
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+	sdkPath := filepath.Join(cwd, "../../sdk/python/env/src")
```

Comment:
This seems quite brittle.  I guess `ProgramTest` usually does this for us, could we use it, using the manual test lifecycle here instead of manually doing this?

Reply:
The test just above has something similar, `filepath.Join("..", "..", "sdk", "python", "env", "src"),` which is passes to program test.

However ProgramTest uses `-e` to install the package, which doesn't work with type checkers (see comment).